### PR TITLE
STM32: Add I2C timing to the F7 and H7 board profile

### LINKED ISSUE #2971: STM32: fix the I2C frequency for H7 and F7
The STM32 port is currently using an erroneous value for the I2C Timing setting on these MCUs, resulting in a constant speed of 100k rather than the proper 400k default with adjustability to 100k. These should be replaced with a proper calculation based on the frequency parameter.

### PR DESCRIPTION
**Pull request by hierophect**
Currently, most F7 and H7 boards have a static and incorrect value for their I2C frequency. This PR adds specific timing values for the F7 and H7 families, in accordance with their maximum possible clock speed as per Circuitpython standard. 

Edits: as per Scott's request, SoC level defaults and board-level overriding of the I2C timing values has been removed. Original PR text below. 

Resolves #2971

~~This PR adds new I2C timing definitions to the board profiles of all H7 and F7 MCUs, ensuring they all meet the 400k standard for Circuitpython. These values should be obtained by board authors from the CubeMX program, which has a utility to best-fit this value based on the MCU type, clock speeds, and desired I2C frequency (standard or fast mode).~~

~~While these macros are a bit of a copout from calculating the timing values manually, finding best-fit timing is an [involved process](https://github.com/stm32duino/Arduino_Core_STM32/blob/684c3e8a574d46b888d7e7fe82399a95f1938d7a/libraries/Wire/src/utility/twi.c#L383) and a bit excessive for the number of dev boards we support. This PR can be seen as a temporary fix, pending an increase in the number of F7 and H7 boards, or a desire for non-standard (not 100k or 400k) I2C frequency support.~~

### CODE REVIEW

File: ports/stm/common-hal/busio/I2C.c [hunk 1]
```diff
@@ -120,7 +120,13 @@ void common_hal_busio_i2c_construct(busio_i2c_obj_t *self,
 
     // Handle the HAL handle differences
     #if (CPY_STM32H7 || CPY_STM32F7)
-    self->handle.Init.Timing = 0x40604E73; //Taken from STCube examples
+    if (frequency == 400000) {
+        self->handle.Init.Timing = CPY_I2CFAST_TIMINGR;
+    } else if (frequency == 100000) {
+        self->handle.Init.Timing = CPY_I2CSTANDARD_TIMINGR;
+    } else {
+        mp_raise_ValueError(translate("MCU supports only I2C Standard and Fast modes"));
```
[the remark this answers is not in the document]
Comment (tannewt) [reply ports/stm/common-hal/busio/I2C.c:128]: Either is fine with me. Up to you.

**Comment by hierophect** [comment]
@tannewt this should be ready for another look

File: ports/stm/boards/stm32f746g_discovery/mpconfigboard.h [hunk 2]
```diff
@@ -42,6 +50,10 @@
 #define CPY_CLK_FLASH_LATENCY (FLASH_LATENCY_6)
 #define CPY_CLK_USB_USES_AUDIOPLL (1)
 
+// Obtain I2C timing values for F7 and H7 boards from ST CubeMX
+#define CPY_I2CFAST_TIMINGR     0x00401959
+#define CPY_I2CSTANDARD_TIMINGR 0x00C0EAFF
```
[the remark this answers is not in the document]
Comment (tannewt) [reply ports/stm/boards/stm32f746g_discovery/mpconfigboard.h:55]: Is it possible to get the bus frequency in the code and switch between the known values? I don't think this should be something a board has to worry about.

**Comment by hierophect** [comment]
@tannewt @k0d this should be ready for a final check-over. As per Scott's request, the I2C timing overrides for the F746 Discovery have been removed.

**Pull request merged**